Web Quiz Engine (Kotlin)/task/src/engine/entity/QuizCompletion.kt

```diff
@@ -0,0 +1,28 @@
+package engine.entity
+
+import com.fasterxml.jackson.annotation.JsonFormat
+import engine.QuizCompletionIdentifier
+import jakarta.persistence.*
+import java.time.ZonedDateTime
+
+@Entity
+@Table(name = "completion")
+data class QuizCompletion(
+    @Id
+    @GeneratedValue
+    val id: QuizCompletionIdentifier? = null,
+    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss.SSSXXX")
+    var completedAt: ZonedDateTime? = null,
```

Comment:
Since you never expect this column to be null, I would add a not nullable index on the Table.
This can be done by adding the annotation `@Column(name = "completed_at", nullable = false)`

This would also make your column snake_case instead of camelCase which is the better practice with SQL.

Reply:
Oh right, usefull, thks.
Done.

> This would also make your column snake_case instead of camelCase which is the better practice with SQL.

True that, have to get into the habit